### PR TITLE
Returning outputs only when asked for for MaskFormer.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Change the return output from `()` to `None` which seems more aligned with
the rest of the library.

Also `auxiliary_logits` seem optional and don't seem to be used by the feature
extractor, so this PR makes them optional too.

Note: I couldn't test the modeling tests, there seem to be no fast tests, and the slow tests are failing for reasons seemingly unrelated to this PR.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/maskformer/modeling_maskformer.py [hunk 1]
```diff
@@ -2421,6 +2428,7 @@ def forward(
         mask_labels: Optional[Tensor] = None,
         class_labels: Optional[Tensor] = None,
         pixel_mask: Optional[Tensor] = None,
+        output_auxiliary_logits: Optional[bool] = None,
```
Comment (sgugger) [inline src/transformers/models/maskformer/modeling_maskformer.py:2431]: This should probably go with a config argument (like `output_hidden_states` etc) that would give a default. Otherwise defaulting to `None` makes no sense.

**Review by sgugger — APPROVED** [review]
Thanks for fixing this!

File: src/transformers/models/maskformer/modeling_maskformer.py (hunk 1 above)
Comment (Narsil) [reply src/transformers/models/maskformer/modeling_maskformer.py:2431]: Ok, I made the change but will wait for @FrancescoSaverioZuppichini advice to make sure this is sound.

Comment (FrancescoSaverioZuppichini) [reply src/transformers/models/maskformer/modeling_maskformer.py:2431]: Thanks! Just a little bit of context, the `auxiliary_logits` are consumed by the loss if `use_auxiliary_loss` was set to true so we could remove them from the output or add a flag in the config to return them

Comment (Narsil) [reply src/transformers/models/maskformer/modeling_maskformer.py:2431]: So if someone uses both `output_auxiliary_logits=True` and `use_auxiliary_loss=True` then it wouldn't work currently is that it ?  (If so we need to change the consuming behavior into a copy I think)

Comment (Narsil) [reply src/transformers/models/maskformer/modeling_maskformer.py:2431]: Friendly ping @FrancescoSaverioZuppichini

Comment (FrancescoSaverioZuppichini) [reply src/transformers/models/maskformer/modeling_maskformer.py:2431]: It will work :)

**Review by FrancescoSaverioZuppichini — APPROVED** [review]
Thanks! LGTM
